File: AtCoder/Beginner/b_081/D_NonDecreasing.go
package main

import "fmt"

func main() {
	var n int
	fmt.Scan(&n)

	maxAbsIndex := 0

	a := make([]int, n)
	for i := range a {
		fmt.Scan(&a[i])
		if Abs(a[i]) > Abs(a[maxAbsIndex]) {
			maxAbsIndex = i
		}
	}

	fmt.Println(2*n - 1)

	for i := range a {
		fmt.Println(maxAbsIndex+1, i+1)
	}

	if a[maxAbsIndex] < 0 {
		for i := n; i > 1; i-- {
			fmt.Println(i, i-1)
		}
	} else {
		for i := range a[:len(a)-1] {
			fmt.Println(i+1, i+2)
		}
	}
}

func Abs(n int) int {
	if n < 0 {
		return -n
	}
	return n
}

File: AtCoder/Beginner/b_081/D_NonDecreasing.scala
import scala.io.StdIn._

object Main extends App {
  val n = readInt()
  val a = readLine().split(" ").map(_.toInt)
  val maxAbsIndex = a.indexOf(a maxBy math.abs)

  println(2 * n - 1)
  for (i <- 1 to a.length) println(s"${maxAbsIndex+1} $i")

  if (a(maxAbsIndex) < 0)
    for (i <- n to 2 by -1) println(s"$i ${i - 1}")
  else
    for (i <- 1 until a.length) println(s"$i ${i + 1}")
}
